Patch by David Halls (david.halls@citrix.com) so connection
isn't closed if just doing a HEAD and Content-Length isn't
returned.
*** curl-7.19.7.orig/lib/transfer.c	2010-10-13 17:30:39.299271715 +0100
--- curl-7.19.7/lib/transfer.c	2010-10-13 17:27:56.415238452 +0100
***************
*** 909,915 ****
        else {
          k->header = FALSE; /* no more header to parse! */
  
!         if((k->size == -1) && !k->chunk && !conn->bits.close &&
             (conn->httpversion >= 11) ) {
            /* On HTTP 1.1, when connection is not to get closed, but no
               Content-Length nor Content-Encoding chunked have been
--- 909,915 ----
        else {
          k->header = FALSE; /* no more header to parse! */
  
!         if((k->size == -1) && !k->chunk && !conn->bits.close && !data->set.opt_no_body &&
             (conn->httpversion >= 11) ) {
            /* On HTTP 1.1, when connection is not to get closed, but no
               Content-Length nor Content-Encoding chunked have been
